### PR TITLE
[LOGIN] - Authentication Endpoint - OT198-25

### PR DESCRIPTION
**Pull request by nicobes14**
[OT198-25](https://alkemy-labs.atlassian.net/browse/OT198-25)
## Description
AS user
I WANT to log into the system
TO access its functionalities

### Criteria of acceptance:

- A request of type POST /auth/login must be made
- The sending of the email and password fields in the request must be validated, and the password must be decrypted to authenticate
- Validate email and password on request
- Check if there is a user with the requested email
- If exists, compare encrypted password with password sent on request
- Return the user if it exists and the password is valid, if an error occurs, return {ok: false}
### Evidence
- happy path
image
- missing arguments
image
- user not found or incorrect password
image

### CODE REVIEW

**Review by honeybadger2788 — APPROVED** [review]
Excelente, Nico

File: services/user.js [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
+        password: bcrypt.hashSync(password, 12),
+      })
+      return user
+    } catch (error) {
+      throw new Error(error)
+    }
+  },
+  tryLogin: async ({ email, password }) => {
+    try {
+      const user = await User.findOne({ where: { email } })
```
Comment (gabodmc) [inline services/user.js:23]: usar el id como identificador de usuario

File: services/user.js [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+        lastName,
+        email,
+        password: bcrypt.hashSync(password, 12),
+      })
+      return user
+    } catch (error) {
+      throw new Error(error)
+    }
+  },
+  tryLogin: async ({ email, password }) => {
```
Comment (gabodmc) [inline services/user.js:21]: aca usaria un getUserById o algo por el estilo. Este nombre no esta muy relacionado a lo que hace el servicio

File: services/user.js [hunk 3]
```diff
@@ -1,22 +1,42 @@
...
+    try {
+      const user = await User.findOne({ where: { email } })
+      if (!user || !bcrypt.compareSync(password, user.password)) {
+        return {
+          code: 401,
+          status: false,
+          message: 'User not found or password incorrect',
```
Comment (gabodmc) [inline services/user.js:28]: cambiar por "invalid credentials" para no dar mucha explicacion de lo que falló

**Review by gabodmc — CHANGES_REQUESTED** [review]
Algunas correciones para hacerle

**Review by JoaquinBar1 — APPROVED** [review]
Muy bueno Nico!

**Review by JoaquinBar1 — APPROVED** [review]

**Review by gabodmc — APPROVED** [review]
Genial Nico!

**Pull request merged**